api/service/legacysync/helpers.go

```diff
@@ -58,39 +58,39 @@ func createSyncConfig(syncConfig *SyncConfig, peers []p2p.Peer, shardID uint32)
 	}
 	// limit the number of dns peers to connect
 	randSeed := time.Now().UnixNano()
-	peers = limitNumPeers(peers, randSeed)
+	targetSize, peers := limitNumPeers(peers, randSeed)
 
 	utils.Logger().Debug().
-		Int("len", len(peers)).
+		Int("peers count", len(peers)).
+		Int("target size", targetSize).
 		Uint32("shardID", shardID).
 		Msg("[SYNC] CreateSyncConfig: len of peers")
 
-	if len(peers) == 0 {
+	if targetSize == 0 {
 		return syncConfig, errors.New("[SYNC] no peers to connect to")
 	}
 	if syncConfig != nil {
 		syncConfig.CloseConnections()
 	}
 	syncConfig = NewSyncConfig(shardID, nil)
 
-	var wg sync.WaitGroup
+	var connectedPeers int
 	for _, peer := range peers {
-		wg.Add(1)
-		go func(peer p2p.Peer) {
-			defer wg.Done()
-			client := downloader.ClientSetup(peer.IP, peer.Port)
-			if client == nil {
-				return
-			}
-			peerConfig := &SyncPeerConfig{
-				ip:     peer.IP,
-				port:   peer.Port,
-				client: client,
-			}
-			syncConfig.AddPeer(peerConfig)
-		}(peer)
+		client := downloader.ClientSetup(peer.IP, peer.Port)
+		if client == nil || !client.IsReady() {
+			continue
+		}
+		peerConfig := &SyncPeerConfig{
+			ip:     peer.IP,
+			port:   peer.Port,
+			client: client,
+		}
+		syncConfig.AddPeer(peerConfig)
+		connectedPeers++
+		if connectedPeers >= targetSize {
```

Comment:
Do we really need new variable when we can just call smth like 
`syncConfig.connectedPeersCount()` ?

Reply:
I think you mean using syncConfig.PeersCount(). right? I should say it also uses mutex and for sure is slower, so, in my opinion in terms of memory usage and performance this variable makes more sense. Any concern you see about this variable?